src/typeInference/annotateLogicalOperator.ts

```diff
@@ -0,0 +1,33 @@
+import { SequenceMultiplicity, SequenceType, ValueType } from '../expressions/dataTypes/Value';
+import astHelper, { IAST } from '../parsing/astHelper';
+
+export function annotateLogicalOperator(ast: IAST): SequenceType {
+	switch (ast[0]) {
+		case 'andOp':
+			return annotateAndOperator(ast);
+		case 'orOp':
+			return annotateOrOperator(ast);
+	}
```

Comment:
I like the sub switch tree that covers the specific cases in here. The annotateAST switch cases are growing into something a bit scary already.

Reply:
I agree that it keeps the annotateAst file cleaner but it does feel a bit redundant. Not sure if can be solved easily though

Reply:
We can later on always predict the type of operator by the length of the ast at that point.